test/framework/src/main/resources/log4j2-test.properties

```diff
@@ -12,7 +12,7 @@
 appender.console.type = Console
 appender.console.name = console
 appender.console.layout.type = PatternLayout
-appender.console.layout.pattern = [%d{ISO8601}][%-5p][%-25c{1.}] [%test_thread_info]%marker %m%n
+appender.console.layout.pattern = [%d{ISO8601}][%-5p][%-25c{1.}] [%test_thread_info]%marker [Thread %tid] %m%n
```

Comment:
@dreamer-89 I think Thread Id is not very useful without thread name (`%tn`), I would even argue thread name is more helpful since it usually includes thread pool as well as thread number. What do you think?

Reply:
Thank you @reta for the review and suggestion. Yes, this makes sense. Also, I see `%test_thread_info` is filtered out part of thread name. It uses node name for engine and test method name for test threads respectively(sample logs below); so it can be removed. Also, removing `Thread` prefix as it is implicit. 

```
[2022-12-25T13:28:27,329][DEBUG][o.o.b.SystemCallFilter   ] [[SUITE-SegmentReplicationRelocationIT-seed#[8D4E3FF132EF9F74]]] [SUITE-SegmentReplicationRelocationIT-seed#[8D4E3FF132EF9F74]] BSD RLIMIT_NPROC initialization successful
[0004-12-25T14:28:32,961][TRACE][o.o.c.r.a.a.BalancedShardsAllocator] [node_t0] [opensearch[node_t0][clusterManagerService#updateTask][T#1]] Start allocating unassigned shards
[0004-12-25T14:28:38,584][INFO ][o.o.i.r.SegmentReplicationRelocationIT] [testPrimaryRelocationWithoutFlushBeforeRelocation] [TEST-SegmentReplicationRelocationIT.testPrimaryRelocationWithoutFlushBeforeRelocation-seed#[8D4E3FF132EF9F74]] --> relocate the shard
```

Reply:
@reta : Ping

Reply:
@dreamer-89 apologies for delay (was off for a few days), I think we (I definitely) missed that the thread conversions are performed by `test_thread_info` marker (and consequently, by `org.opensearch.common.logging.TestThreadInfoPatternConverter` - the change should probably happen right there, keeping the log pattern unchanged.

Reply:
@reta : Updated, please have a look.

Reply:
Thanks @dreamer-89 , LGTM!